src/heap/sharedMem.ml

```diff
@@ -1156,36 +1159,38 @@ module NewAPI = struct
 
   let file_patterns file = read_addr (get_heap ()) (patterns_addr file)
 
-  let cjs_module_info m = read_addr (get_heap ()) (m + 1)
+  let cjs_module_info m = read_addr (get_heap ()) (addr_offset m 1)
 
-  let cjs_module_exports m = read_addr (get_heap ()) (m + 2)
+  let cjs_module_exports m = read_addr (get_heap ()) (addr_offset m 2)
 
-  let cjs_module_type_exports m = read_addr (get_heap ()) (m + 3)
+  let cjs_module_type_exports m = read_addr (get_heap ()) (addr_offset m 3)
 
-  let es_module_info m = read_addr (get_heap ()) (m + 1)
+  let es_module_info m = read_addr (get_heap ()) (addr_offset m 1)
 
-  let es_module_exports m = read_addr (get_heap ()) (m + 2)
+  let es_module_exports m = read_addr (get_heap ()) (addr_offset m 2)
 
-  let es_module_type_exports m = read_addr (get_heap ()) (m + 3)
+  let es_module_type_exports m = read_addr (get_heap ()) (addr_offset m 3)
 
   (* write *)
 
+  (* Read int64 from given byte offset in the heap. This is not bounds checked. *)
+  external unsafe_write_int64 : heap -> int -> int64 -> unit = "%caml_bigstring_set64u"
+
```

Comment:
Shall we provide both `safe` and `unsafe` version? Is this used in some hot paths?

Reply:
Oh, missed this question. I was using the unsafe variant before (Array1.unsafe_set) so this choice is mainly just behavior preserving. The code paths that lead to this call already have assertions that make eliminating the bounds check safe, which is why I used an unsafe set operation. In hindsight, I should have just used the bounds checked operation, since it's almost certainly not the bottleneck here.